src/RequestService.php

```diff
@@ -106,6 +112,10 @@ protected function getAuthAssertion(ApiContextInterface $context): ?string
             return null;
         }
 
+        if (!$context->getMerchantId()) {
+            throw new \LogicException(ApiContextInterface::class . ' is flagged as third party, but is missing a merchant id.');
```

Comment:
don't see the benefit of printing a FQDN here. Not even specific to the implementing class